scripts/collect_ad_boilerplate.py

```diff
@@ -0,0 +1,200 @@
+            break
+    logging.debug(f'Found template dir: {template_dir}')
+
+    # Create a list of file paths to all template files.
+    template_files = []
+    for file in os.listdir(template_dir):
+        if file.endswith('.template'):
+            template_files.append(os.path.join(template_dir, file))
+            logging.debug(f'Found template file {file}')
+
+    # Dict mapping pv name to appropriate EPICS signal, based on typical
+    # PV and PV_RBV format for areaDetector
+    pv_to_signal_mapping = {}
+    include_file_base = False
+
+    for file in template_files:
+        logging.debug(f'Collecting pv info from {os.path.basename(file)}')
+        fp = open(file, 'r')
+
+        lines = fp.readlines()
+        for line in lines:
+
+            # If NDFile.template is included, we need to extend FileBase as well.
+            if line.startswith('include "NDFile.template"'):
+                logging.debug(f'Driver AD{dev_name} uses the NDFile.template file.')
+                include_file_base = True
+
+            # identify any lines that start with 'record
+            if line.startswith('record'):
+                # Get the name of the PV.
+                pv_name = line.split(')')[2][:-1]
+
+                # Check if it is a readback PV
+                if pv_name.endswith('_RBV'):
+                    # If it has a partner PV, switch the signal to SignalWithRBV
+                    if pv_name[:-4] in pv_to_signal_mapping.keys():
+                        logging.debug(f'Identified {pv_name[:-4]} as a record w/ RBV')
+                        pv_to_signal_mapping[pv_name[:-4]] = 'SignalWithRBV'
+                    # Otherwise, it is a read only PV, so use EpicsSignalRO
+                    else:
+                        logging.debug(f'Identified read-only record {pv_name}')
+                        pv_to_signal_mapping[pv_name] = 'EpicsSignalRO'
+                else:
+                    # Otherwise, use the default EpicsSignal
+                    logging.debug(f'Found record {pv_name}')
+                    pv_to_signal_mapping[pv_name] = 'EpicsSignal'
+
+        fp.close()
+
+    return pv_to_signal_mapping, include_file_base
+
+
+def write_cam_class(boilerplate_file, pv_to_signal_mapping, include_file_base, dev_name, det_name, cam_name):
+    """
+    Function that writes the boilerplate cam class. This includes the default configuration
+    attributes, along with all the attributes extracted from the template file.
+
+    This function is not perfect, as extracting a good attribute name from a PV name is not
+    always straightforward, since PV names are far from standard. Currently, the solution simply
+    adds an _ character before any capital letter, and then makes everything lowercase.
+    This works in most cases, but may require minor edits to certain names.
+
+    Examples:
+
+    EnableCallbacks -> enable_callbacks
+    EVTLoadGainFile -> e_v_t_load_gain_file
```

Comment:
Can the [`inflection`](https://inflection.readthedocs.io/en/latest/) library be used for such purposes? Will require adding it into `requirements.txt`.

Example:
```py
In [1]: import inflection

In [2]: inflection.underscore('EnableCallbacks')
Out[2]: 'enable_callbacks'

In [3]: inflection.underscore('EVTLoadGainFile')
Out[3]: 'evt_load_gain_file'
```

Reply:
This looks cool, I'd never heard of it before. Seems it would fill the need nicely. Is there any reason we shouldn't add it as a dependency to `requirements.txt`?

Reply:
I think it's fine to add it as a dependency, as we already have a couple of packages. Also, we deploy the `inflection` package on the floor already, so it will not be new in that integral environment.

What do other people think?